### PR TITLE
6353: Add error handler from BCL

### PR DESCRIPTION
**Pull request by BenBJSS**
[AB#6353](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6353)

### CODE REVIEW

File: app/pages/solutions-list/filterType/manifestProvider.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
 export const getSolutionListManifest = (type) => {
   const filePath = `/${type}/manifest.json`;
 
   if (!fs.existsSync(path.join(__dirname, filePath))) {
-    throw new Error(`No manifest found for filter type: ${type}`);
+    throw new ErrorContext({
+      status: 404,
+      description: `No manifest found for filter type: ${type}`,
+    });
```
Comment (alibombaye) [inline app/pages/solutions-list/filterType/manifestProvider.js:12]: This error shouldn't leak to the page.. We should log this error but return a general 500 error.

File: app/pages/error/template.test.js [hunk 2]
```diff
@@ -7,46 +7,41 @@ const setup = {
 };
 
 describe('error page', () => {
-  it('should render the error title', componentTester(setup, (harness) => {
-    const context = { message: 'an error message' };
+  it('should render the error backLink', componentTester(setup, (harness) => {
+    const context = {
+      backLinkText: 'Error backLinkText',
+      backLinkHref: 'http://errorBackLinkHref.com',
+    };
```
Comment (kat1906) [inline app/pages/error/template.test.js:14]: do we need tests for the default, ie when the context is not given?

Comment (BenBJSS) [reply app/pages/error/template.test.js:14]: I don't think so, what if we changed them in BCL? We'd have to change them in all the apps

File: app/pages/solutions-list/filterType/manifestProvider.js (hunk 1 above)
Comment (alibombaye) [reply app/pages/solutions-list/filterType/manifestProvider.js:11]: Can we still log.error that the manifest wasn't found

**Pull request merged**